cpp/src/parquet/encryption/encryption_internal.cc

```diff
@@ -758,9 +795,14 @@ void QuickUpdatePageAad(int32_t new_page_ordinal, std::string* AAD) {
   std::memcpy(AAD->data() + AAD->length() - 2, page_ordinal_bytes.data(), 2);
 }
 
-void RandBytes(unsigned char* buf, int num) {
+void RandBytes(unsigned char* buf, size_t num) {
+  if (num > static_cast<size_t>(std::numeric_limits<int>::max())) {
+    std::stringstream ss;
+    ss << "Length " << num << " for RandBytes overflows int";
+    throw ParquetException(ss.str());
+  }
   openssl::EnsureInitialized();
-  RAND_bytes(buf, num);
+  RAND_bytes(buf, static_cast<int>(num));
```

Comment:
a minor question, what would `num == 0`?

Reply:
This seems to work fine from testing it.

Looking at the [RAND_bytes documentation](https://docs.openssl.org/3.3/man3/RAND_bytes/) though, I notice that it says that the return value should be checked as it's not always guaranteed that random bytes can be generated.

I think this should probably be fixed but maybe in a separate PR?

Reply:
Ok cool, I've added the check now